local-tests/chainrunner/chain.py

```diff
@@ -21,32 +22,48 @@ def __init__(self, workdir):
         os.makedirs(workdir, exist_ok=True)
         self.path = op.abspath(workdir)
         self.nodes = []
+        self.validator_nodes = []
+        self.nonvalidator_nodes = []
 
     def __getitem__(self, i):
         return self.nodes[i]
 
     def __iter__(self):
         return iter(self.nodes)
 
-    def bootstrap(self, binary, accounts, **kwargs):
-        """Bootstrap the chain. `accounts` should be a list of strings.
+    def bootstrap(self, binary, validator_accounts, accounts=None, **kwargs):
+        """Bootstrap the chain. `validator_accounts` should be a list of strings.
         Flags `--account-ids`, `--base-path` and `--raw` are added automatically.
         All other flags are taken from kwargs"""
+        accounts = accounts or []
         cmd = [check_file(binary),
                'bootstrap-chain',
                '--base-path', self.path,
-               '--account-ids', ','.join(accounts), '--raw']
+               '--account-ids', ','.join(validator_accounts), '--raw']
         cmd += flags_from_dict(kwargs)
 
         chainspec = op.join(self.path, 'chainspec.json')
         with open(chainspec, 'w', encoding='utf-8') as f:
             subprocess.run(cmd, stdout=f, check=True)
 
-        self.nodes = []
-        for a in accounts:
-            n = Node(binary, chainspec, op.join(self.path, a), self.path)
-            n.flags['node-key-file'] = op.join(self.path, a, 'p2p_secret')
-            self.nodes.append(n)
+        def account_to_node(account):
+            n = Node(binary, chainspec, op.join(self.path, account), self.path)
+            n.flags['node-key-file'] = op.join(self.path, account, 'p2p_secret')
+            return n
+
+        self.validator_nodes = [account_to_node(a) for a in validator_accounts]
+        self.nonvalidator_nodes = [account_to_node(a) for a in accounts]
+
+        self.nodes = list(chain(self.nonvalidator_nodes, self.validator_nodes))
```

Comment:
Why not `self.nodes = self.nonvalidator_nodes + self.validator_nodes`? Not really an issue, just curious

Reply:
:| changed